blockscout-ens/bens-logic/src/subgraphs_reader/pagination/paginator.rs

```diff
@@ -0,0 +1,130 @@
+use serde::Deserialize;
+use std::fmt::Display;
+
+pub trait Paginator<I> {
+    fn add_to_query(&self, query: &mut sea_query::SelectStatement) -> Result<(), anyhow::Error>;
+
+    fn paginate_result(&self, items: Vec<I>) -> Result<PaginatedList<I>, anyhow::Error>;
+}
+
+#[derive(Debug, Clone)]
+pub struct PaginationInput<S> {
+    pub sort: S,
+    pub order: Order,
+    pub page_size: u32,
+    pub page_token: Option<String>,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct PaginatedList<I> {
+    pub items: Vec<I>,
+    pub next_page_token: Option<String>,
+}
+
+impl<I> PaginatedList<I> {
+    pub fn empty() -> Self {
+        Self {
+            items: vec![],
+            next_page_token: None,
+        }
+    }
```

Comment:
Nit: maybe just derive a `Default` trait over the struct?

Reply:
Then we need to define Default for Vec<I> (for example default for Vec<Domain> cannot be derived since Domain doesnt implement Default)
<img width="825" alt="image" src="https://github.com/blockscout/blockscout-rs/assets/41516657/21b6e390-ad1b-4cc8-a89a-4c8cf3c19b49">

Reply:
¯\_(ツ)_/¯